cirq/ops/common_gates.py

```diff
@@ -490,36 +489,85 @@ def measure_each(*qubits: raw_types.QubitId,
-        return gate_features.TextDiagramInfo(('H',))
+    def __init__(self, *,  # Forces keyword args.
+                 half_turns: Optional[Union[value.Symbol, float]] = None,
+                 rads: Optional[float] = None,
+                 degs: Optional[float] = None) -> None:
+        """Initializes the gate.
+
+        At most one angle argument may be specified. If more are specified,
+        the result is considered ambiguous and an error is thrown. If no angle
+        argument is given, the default value of one half turn is used.
+
+        Args:
+            half_turns: The relative phasing of H's eigenstates, in half_turns.
+            rads: The relative phasing of H's eigenstates, in radians.
+            degs: The relative phasing of H's eigenstates, in degrees.
+        """
+        super().__init__(exponent=value.chosen_angle_to_half_turns(
+            half_turns=half_turns,
+            rads=rads,
+            degs=degs))
+
+    def _canonical_exponent_period(self) -> Optional[float]:
+        return 2
+
+    def _with_exponent(self,
+                       exponent: Union[value.Symbol, float]) -> 'HGate':
+        return HGate(half_turns=exponent)
+
+    def _eigen_components(self):
+        component0 = np.divide(np.array([[(3 + 2 * np.sqrt(2)),
+                                          (1 + np.sqrt(2))],
+                                         [(1 + np.sqrt(2)), (1)]]),
+                               2 * (2 + np.sqrt(2)))
+
+        component1 = np.divide(np.array([[(3 - 2 * np.sqrt(2)),
+                                          (1 - np.sqrt(2))],
+                                         [(1 - np.sqrt(2)), (1)]]),
+                               2 * (2 - np.sqrt(2)))
+
+        return [(0, component0), (1, component1), ]
+
+    @property
+    def half_turns(self) -> Union[value.Symbol, float]:
+        return self._exponent
 
     def default_decompose(self, qubits):
         q = qubits[0]
-        yield Y(q)**0.5
-        yield X(q)
+
+        if self._exponent == 1:
+            yield Y(q)**0.5, X(q)
+            return
+
+        yield Y(q)**0.25
+        yield X(q)**self.half_turns
+        yield Y(q)**-0.25
 
     def inverse(self):
         return self
 
-    def matrix(self):
-        """See base class."""
-        s = math.sqrt(0.5)
-        return np.array([[s, s], [s, -s]])
+    def text_diagram_info(self, args: gate_features.TextDiagramInfoArgs
+                          ) -> gate_features.TextDiagramInfo:
+        return gate_features.TextDiagramInfo(('H',))
 
     def known_qasm_output(self,
                           qubits: Tuple[raw_types.QubitId, ...],
                           args: gate_features.QasmOutputArgs) -> Optional[str]:
         args.validate_version('2.0')
-        return args.format('h {0};\n', qubits[0])
+        if self.half_turns == 1:
+            return args.format('h {0};\n', qubits[0])
+        else:
+            return args.format('ry({0:half_turns}) {3};\nrx({1:half_turns})'
```

Comment:
I would break the strings after the `\n` like so:

```python
return args.format('ry({0:half_turns}) {3};\n'
                   'rx({1:half_turns}) {3};\n'
                   ...
```